### PR TITLE
Leap year with TMY weather file

### LINKED ISSUE #961: Review leap year handling
We are currently assuming there are 366 vs 365 days in a year based on [this code](https://github.com/NREL/OpenStudio-HPXML/blob/c72d748c28e3d9f73021d6ce1f9847169cca2f76/HPXMLtoOpenStudio/resources/constants.rb#L20). We were previously using [this code](https://github.com/NREL/OpenStudio-HPXML/blob/fdeaba880133afb1fc62957fdf4bf227a99dce89/HPXMLtoOpenStudio/resources/schedules.rb#L74), which I assume was identical logic.

However, just having a calendar year specified that is a leap year is not sufficient. One also needs to supply a weather file that has 8784 hours in it, otherwise the simulation has 365 days. The issue can be seen with the `base-simcontrol-calendar-year-custom.xml` file -- it has the calendar year set at 2012, so it's assumed to have a leap day, but the weather file (and thus simulation results) has 8760 hours. It's not clear what effect this issue has on annual/timeseries output, but it's presumably pretty minor.

Found while working with the Cambium dataset and testing leap years.

cc @joseph-robertson

### PR DESCRIPTION
**Pull request by joseph-robertson**
## Pull Request Description

Closes #961.

Checklist:
- [x] Changes `base-simcontrol-calendar-year-custom.xml` calendar year to 2010 (non-leap year)
- [x] Updates `location.rb` to throw error if leap year is specified with weather file that is not 8784 hours
- [x] Creates test case in `test_measure_error_messages` for throwing this error

## Checklist

Not all may apply:

- [ ] ~EPvalidator.xml has been updated~
- [x] Tests (and test files) have been updated
- [x] Documentation has been updated
- [x] Changelog has been updated
- [x] `openstudio tasks.rb update_measures` has been run
- [x] No unexpected regression test changes on CI (checked comparison artifacts)

### CODE REVIEW

File: HPXMLtoOpenStudio/resources/location.rb [hunk 1]
```diff
@@ -40,7 +40,11 @@ def self.apply_site(model, epw_file)
     site.setElevation(epw_file.elevation)
   end
 
-  def self.apply_year(model, hpxml)
+  def self.apply_year(model, hpxml, epw_file)
+    if !epw_file.startDateActualYear.is_initialized && Date.leap?(hpxml.header.sim_calendar_year)
+      fail "Specified a leap year (#{hpxml.header.sim_calendar_year}) with a TMY weather file. A non-leap year, or an AMY weather file with the same actual year, must be specified."
+    end
```
Comment (shorowit) [inline HPXMLtoOpenStudio/resources/location.rb:46]: I tend to feel that this check is too specific. Yes, a TMY file should have 8760 hours, but an AMY file that someone developed could also have 8760 hours.

How would you feel about this? (Completely untested.)

```suggestion
    n_hours = epw_file.data.size
    if n_hours != 8784 && Date.leap?(hpxml.header.sim_calendar_year)
      fail "Specified a leap year (#{hpxml.header.sim_calendar_year}) but weather data has #{n_hours} hours."
    end
```

File: workflow/tests/base_results/results.csv [hunk 2]
```diff
@@ -316,7 +316,7 @@ base-schedules-detailed-smooth.xml,34.99,34.99,14.52,0.0,0.0,0.0,0.0,0.0,0.0,0.3
...
+base-simcontrol-calendar-year-custom.xml,34.98,34.98,14.53,0.0,0.0,0.0,0.0,0.0,0.0,0.38,3.49,0.67,9.16,0.0,0.0,4.51,0.0,0.33,0.0,0.0,0.0,0.0,2.22,0.0,0.0,0.32,0.37,1.51,1.53,0.0,2.11,8.37,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,14.53,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,13.73,11.33,9.23,0.61,0.0,0.0,0.0,0.0,2140.0,3451.0,20.46,20.7,0.0,3.89,3.98,0.54,4.0,0.71,-0.09,0.0,0.0,3.07,-0.11,5.01,0.0,0.76,0.0,3.33,-11.46,0.0,-0.02,-0.45,-0.01,3.86,-0.02,8.93,0.0,0.0,-5.98,-0.07,-1.04,-6.54,-0.15,0.0,3.01,9.83,1355.0,998.0,11400.0,2616.0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0
```
Comment (shorowit) [inline workflow/tests/base_results/results.csv:319]: I assume these small diffs are due to the Day of Week for Start Day in the simulation changing?

**Review by shorowit — COMMENTED** [review]
I am comfortable with the general approach here but suggest one change related to the error checking/message.

**Comment by shorowit** [comment]
Also please update the PR checklist -- this needs documentation and changelog updates at a minimum.

File: workflow/tests/base_results/results.csv (hunk 2 above)
Comment (joseph-robertson) [reply workflow/tests/base_results/results.csv:319]: Yes I believe so.

**Review by shorowit — APPROVED** [review]

**Pull request merged**